### PR TITLE
Bring (py)Chiquito in

### LINKED ISSUE #82: Moving PyChiquito to the main repo
The python code of pychiquito should be in :

`src/frontend/python/chiquito`

so the python package is called chiquito. And the rust wrapped code in python should be in the python module `chiquito.rust_chiquito`.

The code that is currently in `src/frontend/pychiquito.rs` plus the code in the `https://github.com/qwang98/PyChiquito` repo in `src/lib.rs` should be merged into `src/frontend/python/mod.rs`.

A makefile should be added to the root of the project with a rule to build pychiquito.

The python examples should be moved to the examples folder with the rust examples.

The tutorial should be moved to a `tutorial` folder.

### PR DESCRIPTION
**Pull request by trangnv**
Resolve #82 
- Add `scr/python.rs` contains Python Module
- `frontend/python/chiquito` is home for python-source
- dependencies = ["py_ecc"] included in `pyproject.toml` so users don't have to `pip install py_ecc`
- add `fibonacci.py` to examples

### CODE REVIEW

File: .gitignore
Comment (qwang98) [inline .gitignore:6]: Purely cosmetic but we can add an empty line here for end of file.

**Comment by qwang98** [comment]
This is also to confirm with @leolara, but my understanding is that Chiquito Python code (non examples) should be in `src/frontend/python/chiquito` rather than `frontend/python/chiquito`? Then `pychiquito.rs` and maturin bindings in `python.rs` in this branch should go into `src/frontend/python`.

File: .gitignore
Comment (qwang98) [inline .gitignore:6]: Entirely cosmetic but we typically add one empty line to end of file.

File: README.md
Comment (qwang98) [inline README.md:66]: Thanks for adding this!

Are "install chiquito with pip" and "build from source" considered two different methods for installation or two steps of installation?

To confirm, after `pip install chiquito`, is `rust_chiquito` directly accessible from `chiquito.rust_chiquito` or requires `maturin develop`? My guess is the latter.

File: examples/fibonacci.py
Comment (qwang98) [inline examples/fibonacci.py:1]: Note that I've updated the example in the PyChiquito repo, but we can focus on getting the structure working before porting over the latest PyChiquito updates.

File: src/python.rs
Comment (qwang98) [inline src/python.rs:1]: Again, my understanding is that this file plus `pychiquito.rs` should go into `src/frontend/python/mod.rs`

File: examples/fibonacci.py
Comment (qwang98) [reply examples/fibonacci.py:1]: Note to self: there are also some python comments that I should delete once we have this merged.

File: .gitignore
Comment (trangnv) [reply .gitignore:6]: cosmetic haha

File: README.md
Comment (trangnv) [reply README.md:66]: It's either to install, pip package will be available for python user, `maturin develop` is only for who want to build from source/develop. 
Will make it clear

**Comment by trangnv** [comment]
Let wait for Leo to reach the agreement for the structure then I will reorg accordingly.

File: src/python.rs
Comment (leolara) [reply src/python.rs:1]: Yes, that is the spec of the task, if that is possible, i would like it to be there

File: .gitignore
Comment (leolara) [reply .gitignore:6]: Actually, in some edge cases it can lead to unnecessary git conflicts when several people changes the last lines of the file. So we have this rule of always ending all files with an empty line

**Comment by leolara** [comment]
the content of frontend/pychiquito.rs together with this PR src/pyton.rs should be in frontend/python/mod.rs

**Comment by leolara** [comment]
> This is also to confirm with @leolara, but my understanding is that Chiquito Python code (non examples) should be in `src/frontend/python/chiquito` rather than `frontend/python/chiquito`? Then `pychiquito.rs` and maturin bindings in `python.rs` in this branch should go into `src/frontend/python`.

Yes, when I have said frontend/python, I meant src/frontend/python

**Comment by leolara** [comment]
It was my mistake, I put it wrong in the issue, I have corrected it now.

**Comment by qwang98** [comment]
FYI, we should also add a `tutorial` folder that contains all the Jupyter files.

**Comment by leolara** [comment]
> FYI, we should also add a `tutorial` folder that contains all the Jupyter files.

:+1:

**Comment by trangnv** [comment]
Dont merge yet but please see how it look

**Comment by leolara** [comment]
@trangnv we usually let the contributor to merge once approved

**Comment by leolara** [comment]
@trangnv I think it is missing:
 + move src/frontend/pychiquito.rs content to src/frontend/python/mod.rs ;  src/frontend/pychiquito.rs should be removed
 + move jupyter tutorials
 + Perhaps add a `make build` that builds the rust and the python

**Review by leolara — APPROVED** [review]
Excellent!!

**Comment by leolara** [comment]
@trangnv It is approved, merge it please

**Comment by trangnv** [comment]
> @trangnv It is approved, merge it please

👍

**Pull request merged**
